### PR TITLE
Show areas in deck cards upon opening the page + refactoring relevant code

### LINKED ISSUE #1134: Areas on deckCard don't show until after a deckCard is clicked
**Describe the bug**
Areas on deckCard don't show until after a deckCard is clicked

**To Reproduce**
1. Go to flotilla frontpage
2. Note that area string under deck name is missing (if not, reload page)
3. Click on one of the deck cards
4. Note that the area string appears

**Expected behavior**
Areas should show shortly after the page loads

**Screenshots**
Before clicking on main deck:
image

After clicking on main deck:
image

### PR DESCRIPTION
**Pull request by andchiind**
Closes #1134 

This PR also includes refactoring to reduce dependencies between components.

### CODE REVIEW

File: frontend/src/components/Pages/InspectionPage/InspectionUtilities.tsx [hunk 1]
```diff
@@ -134,10 +134,10 @@ export const compareInspections = (inspection1: Inspection, inspection2: Inspect
 
 interface ICardMissionInformationProps {
     deckName: string
-    deckMissions: DeckMissionType
```
Comment (Eddasol) [inline frontend/src/components/Pages/InspectionPage/InspectionUtilities.tsx:137]: DeckMissionType is no longer in use. Remove import

File: frontend/src/components/Pages/InspectionPage/InspectionSection.tsx [hunk 2]
```diff
@@ -71,13 +59,12 @@ export function InspectionSection({ scheduledMissions, ongoingMissions }: IInspe
...
             selectedMissions.forEach((mission) => {
-                if (Object.keys(scheduledMissions).includes(mission.id) && scheduledMissions[mission.id])
-                    setIsAlreadyScheduled(true)
+                if (isOngoing(mission) || isScheduled(mission)) setIsAlreadyScheduled(true)
                 else unscheduledMissions = unscheduledMissions.concat([mission])
             })
             setUnscheduledMissions(unscheduledMissions)
         }
-    }, [isDialogOpen, scheduledMissions, selectedMissions])
+    }, [isDialogOpen, ongoingMissions, missionQueue, selectedMissions])
```
Comment (Eddasol) [inline frontend/src/components/Pages/InspectionPage/InspectionSection.tsx:67]: image

File: frontend/src/components/Pages/InspectionPage/InspectionOverview.tsx [hunk 3]
```diff
@@ -35,29 +35,10 @@ export function InspectionOverviewSection() {
     const { TranslateText } = useLanguageContext()
     const [activeTab, setActiveTab] = useState(0)
     const [allMissions, setAllMissions] = useState<Inspection[]>()
-    const [ongoingMissionsMap, setOngoingMissions] = useState<ScheduledMissionType>({})
-    const [scheduledMissions, setScheduledMissions] = useState<ScheduledMissionType>({})
-    const { ongoingMissions, missionQueue } = useMissionsContext()
```
Comment (Eddasol) [inline frontend/src/components/Pages/InspectionPage/InspectionOverview.tsx:40]: useMissionsContext is no longer in use. Remove import

File: frontend/src/components/Pages/InspectionPage/DeckCards.tsx [hunk 4]
```diff
@@ -43,90 +43,76 @@ export function DeckCards({
...
-                    .map((area) => {
-                        return area.areaName.toLocaleUpperCase()
-                    })
-                    .sort()
-                    .join(' | ')
-                newAreas[deckName] = {
-                    areaString: formattedAreaNames,
-                }
-            })
+    const areasString = deckData.areas
```
Comment (Eddasol) [inline frontend/src/components/Pages/InspectionPage/DeckCards.tsx:46]:
```suggestion
    const formattedAreaNames = deckData.areas
```
I prefer this name

**Review by mrica-equinor — APPROVED** [review]
LGTM :)

**Pull request merged**